typescript/src/entities/variation_field.ts

```diff
@@ -88,4 +90,58 @@ export class VariationField extends Entity {
 
   @VariationField.property({arrayType: "VariationFieldsOption"})
   public options?: Array<VariationFieldsOption>;
+
+  public isSelectable = () => {
+    if (this.fieldType === undefined) {
+      throw new Error("fieldType is undefined, did you forget to embed it?");
+    }
+    const selectable = new Set([FieldType.SELECT,
+                                FieldType.CHECKBOX,
+                                FieldType.RADIO,
+                                FieldType.IMAGE_SELECT,
+                                FieldType.COLOUR_SELECT]);
+    return selectable.has(this.fieldType);
+  }
+
+  public buildEmptyVariation = () => {
+    if (this.defaultValue === undefined) {
+      throw new Error("defaultValue is undefined, did you forget to embed it?");
+    }
+    if (this.variationCost === undefined) {
+      const err = "variationCost is undefined, did you forget to embed it?";
+      throw new Error(err);
+    }
+    if (this.options === undefined) {
+      throw new Error("options is undefined, did you forget to embed it?");
+    }
+    const result = new this.merchi.Variation(this.merchi);
+    if (this.isSelectable()) {
+      let onceOffCost = 0;
+      const value = [];
+      for (const option of this.options) {
+        if (option.default) {
+          if (option.variationCost === undefined) {
+            throw new Error("option.variationCost is undefined, did you " +
```

Comment:
same suspection why do embed check on primary value

Reply:
OK, the error suggestion is bad, it's unlikely to be from a missed embed.

the situation will more likely arise due creating a new object that hasn't fetched anything at all from the server yet.